src/components/TopBar/Breadcrumb.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+
+import React, { PureComponent } from 'react'
+import styled from 'styled-components'
+import type { Account } from '@ledgerhq/live-common/lib/types/account'
+import type { T } from 'types/common'
+import { compose } from 'redux'
+import { connect } from 'react-redux'
+import { createStructuredSelector } from 'reselect'
+import { push } from 'react-router-redux'
+import { translate } from 'react-i18next'
+
+import { accountsSelector } from '../../reducers/accounts'
+import Crumb from './Crumb'
+import Button from '../base/Button'
+
+type Props = {
+  pathname: string,
+  accounts: Account[],
+  push: string => void,
+  t: T,
+}
+
+const mapStateToProps = createStructuredSelector({
+  accounts: accountsSelector,
+})
+
+const mapDispatchToProps = {
+  push,
+}
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  > * {
+    font-family: 'Open Sans';
+    font-weight: 600;
+    font-size: 12px;
+    color: ${p => p.theme.colors.grey};
+  }
+
+  > :first-child {
+    padding: 0px;
+    &:hover {
+      background: transparent;
+      text-decoration: underline;
+    }
+  }
+`
+
+const Separator = styled.div`
+  &::after {
+    content: '/';
+    font-size: 13px;
+    color: ${p => p.theme.colors.fog};
+    padding: 0 15px;
+  }
+`
+
+class Breadcrumb extends PureComponent<Props> {
+  processItemsForDropdown = (items: any[]) =>
+    items.map(item => ({ key: item.id, label: item.id, account: item }))
+
+  render() {
+    const { t, push, pathname, accounts } = this.props
+    let tokenItems
+    let token
+
+    if (!pathname.startsWith('/account/')) return null
+
+    const [, , accountId, tokenId] = pathname.split('/')
+    const account = accounts.find(a => a.id === accountId)
+    const accountsItems = this.processItemsForDropdown(accounts)
+
+    if (tokenId && account && account.tokenAccounts) {
+      const tokens = account.tokenAccounts
+      tokenItems = this.processItemsForDropdown(tokens)
+      token = tokens.find(t => t.id === tokenId)
+    }
+
+    return (
+      <Wrapper>
+        <Button onClick={() => this.props.push('/accounts/')}>{t('accounts.title')}</Button>
+        <Separator />
+        <Crumb push={push} items={accountsItems} account={account} />
```

Comment:
i would suggest we directly pass in the accounts, not sure why we need to map them with label and key from this component point of view.

Reply:
I reused the dropdown component, it seemed to want this, I'll look again